app.py

```diff
@@ -141,6 +133,66 @@ async def create_directory():
         return jsonify(["CreateDirectoryRequestError: " + str(error)]), 400
 
 
+@app.post("/image-validation")
+async def image_validation():
+    """
+    Validates an image based on its extension, header, size, and resizability.
+
+    Returns:
+        A JSON response containing a validator hash.
+
+    Raises:
+        ImageValidationError: If the image fails any of the validation checks.
+    """
+    try:
+
+        data = await request.get_json()
+        image_base64 = data["image"]
+
+        header, encoded_image = image_base64.split(",", 1)
+
+        image_bytes = base64.b64decode(encoded_image)
+
+        image = Image.open(io.BytesIO(image_bytes))
+        image_extension = image.format.lower()
+
+        # extension check
+        if image_extension not in VALID_EXTENSION:
```

Comment:
I suggest we do not rely on extension for validation but on checking actual file header and getting the MIME type from it:

https://pypi.org/project/filemagic/